tools/tkn-results/internal/config/config.go

```diff
@@ -0,0 +1,91 @@
+package config
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+)
+
+var (
+	addr      *string = pflag.StringP("addr", "a", "", "Result API server address")
+	authToken *string = pflag.StringP("authtoken", "t", "", "authorization bearer token to use for authenticated requests")
+
+	env = map[string]string{
+		"TKN_RESULTS_SSL_ROOTS_FILE_PATH":      "Path to local SSL cert to use.",
+		"TKN_RESULTS_SSL_SERVER_NAME_OVERRIDE": "SSL server name override (useful if using with a proxy such as kubectl port-forward).",
+	}
+)
+
+type Config struct {
+	Address string
+	Token   string
+
+	SSL SSLConfig
+}
+
+type SSLConfig struct {
+	RootsFilePath      string `mapstructure:"roots_file_path"`
+	ServerNameOverride string `mapstructure:"server_name_override"`
+}
+
+func init() {
+	viper.SetConfigName("results")
+	viper.SetConfigType("yaml")
+	viper.AddConfigPath("$HOME/.config/tkn")
+}
+
+func GetConfig() (*Config, error) {
+	pflag.Parse()
+	if err := viper.BindPFlags(pflag.CommandLine); err != nil {
+		return nil, err
+	}
+
+	for k := range env {
+		if err := viper.BindEnv(k); err != nil {
+			return nil, err
+		}
+	}
+
+	// Config should be evaluated in the following order (last wins):
+	// 1. Environment variables
+	// 2. Config File
+	// 3. Flags
+
+	// Initial config is contains the env variables,
+	// so that the unmarshal can take priority if those values are set.
+	cfg := &Config{
+		SSL: SSLConfig{
+			RootsFilePath:      viper.GetString("TKN_RESULTS_SSL_ROOTS_FILE_PATH"),
```

Comment:
given env var names like `TKN_RESULTS_SSL_ROOTS_FILE_PATH` appear multiple times, suggest adding them as named `const` declarations for DRY.

Reply:
Done!